### PR TITLE
Fix IBAN conversion validation

### PR DESCRIPTION
**Pull request by gbirke**
Fixes https://phabricator.wikimedia.org/T141381

### CODE REVIEW

**Comment by KaiNissen** [comment]
This reintroduces [phab:T140476](https://phabricator.wikimedia.org/T140476), which should have been fixed by #568, but apparently wasn't.

**Comment by JeroenDeDauw** [comment]
I supposed Kai's comment has been addressed since the commit is more recent now... +1 as I don't see anything obviously wrong, though then I don't understand the code well enough to catch a lot of things.

If this is something that was wrong and then not fully fixed and reintroduced, having a test for it seems real nice. Is that difficult to do nicely in this case?

File: app/js/lib/form_validation.js [hunk 1]
```diff
@@ -97,6 +100,10 @@ var jQuery = require( 'jquery' ),
 				};
 				validationUrl = this.validationUrlForNonSepa;
 			}
+			// avoid sending incomplete data to the server
+			if ( _.find( data, isEmptyString ) !== undefined ) {
```
Comment (JeroenDeDauw) [inline app/js/lib/form_validation.js]: It took me a good number of seconds to fully interpret what this is doing, promoting me to create https://github.com/wmde/FundraisingFrontend/pull/591/commits/f6c2b0298ee96b83292b59116ebca81efa523740. As it is here, you need to deal with not relevant low level details such as the behaviour of the _.find function.

As I found out there, there is no need for such a general check, or for the local variables in this method. https://github.com/wmde/FundraisingFrontend/pull/591/commits/646dcc922e0ff9b4f0c974a2c6868ed27c4c299b

PR: https://github.com/wmde/FundraisingFrontend/pull/591

**Comment by gbirke** [comment]
I think for testing this behavior, you really need a browser test, that simulates a user clicking in the interface.

**Comment by KaiNissen** [comment]
There is one tiny bit missing: If I switch from direct debit to any other payment method, the bank data fields are empty and receive a green validity indicator in all cases (were valid before, were invalid before, were already empty).

**Comment by gbirke** [comment]
Addressed comment from @KaiNissen 

@JeroenDeDauw Could you merge this?

**Pull request merged**